country-a-service/epps-service/src/main/java/dk/sundhedsdatastyrelsen/ncpeh/service/mapping/DispensationMapper.java

```diff
@@ -399,18 +364,18 @@ static DrugType drug(Document cda) throws XPathExpressionException {
     }
 
     static SubstancesType substances(Document cda) throws XPathExpressionException {
-        var ingredientNodes = evalNodeMany(cda, XPaths.activeIngredients);
+        var ingredientNodes = xpath.evalNodeList(XPaths.activeIngredients, cda);
         if (ingredientNodes.isEmpty()) {
             return null;
         }
         var b = SubstancesType.builder();
         for (var node : ingredientNodes) {
-            var substanceName = eval(node, "pharm:ingredientSubstance/pharm:name");
+            var substanceName = xpath.evalString("pharm:ingredientSubstance/pharm:name", node);
             if (StringUtils.isNotBlank(substanceName)) {
                 b.addActiveSubstance().withFreeText(substanceName);
             } else {
-                var substanceCode = eval(node, "pharm:ingredientSubstance/pharm:code/@code");
-                var substanceDisplayName = eval(node, "pharm:ingredientSubstance/pharm:code/@displayName");
+                var substanceCode = xpath.evalString("pharm:ingredientSubstance/pharm:code/@code", node);
+                var substanceDisplayName = xpath.evalString("pharm:ingredientSubstance/pharm:code/@displayName", node);
                 if (StringUtils.isNotBlank(substanceCode)) {
                     b.addActiveSubstance().withFreeText("%s %s".formatted(substanceCode, substanceDisplayName).trim());
```

Comment:
```suggestion
                    b.addActiveSubstance().withFreeText("%s %s".formatted(substanceCode, substanceDisplayName));
```
you've moved the trim to `evalString`

Reply:
Fair enough